lib/corefoundation/exceptions.rb

```diff
@@ -0,0 +1,16 @@
+module CF
+  module Exceptions
+    # Raised when the preference value couldn't be found for a domain.
+    class PreferenceDoesNotExist < RuntimeError
```

Comment:
Inherit from `StandardError`

Reply:
RuntimeError is in fact the subclass of StandardError. RuntimeError seems more appropriate because it suggests that we attempted to do something that's considered invalid which raised this error. In this case, the invalid operation was to read a preference value that doesn't exist.

Reply:
See, defining custom exceptions in ruby and default behaviour of `rescue`

Reply:
I did see it before defining these classes. Could you please elaborate on what point you're trying to make?

Reply:
As I said earlier, I have read about it. And I didn't find any mention of avoiding inheriting from a subclass. In fact, I have also seen a lot of codebases define custom exceptions using `RuntimeError` class, including chef and have also seen a lot of examples from ruby experts in the community using RuntimeError when defining custom exceptions. Moreover, the ancestor chain shouldn't matter in this use case because we simply care about a descriptive custom error.